octoprint_prometheus_exporter/__init__.py

```diff
@@ -263,9 +263,12 @@ def gcodephase_hook(self, comm_instance, phase, cmd, cmd_type, gcode, subcode=No
 			if self.print_progress_label != '':
 				data = self._printer.get_current_data()
 				#self._logger.info(data)
-				self.print_time_elapsed.labels(self.print_progress_label).set(data['progress']['printTime'])
-				self.print_time_est.labels(self.print_progress_label).set(data['job']['estimatedPrintTime'])
-				self.print_time_left_est.labels(self.print_progress_label).set(data['progress']['printTimeLeft'])
+				if data['progress']['printTime'] is not None:
```

Comment:
I'm not a python pro, but if we defensive program out this; `data['progress']['printTime']` shouldn't we defensive program out `data['progress']` too?
Like `if (data['progress'] is not None) && (data['progress']['printTime'] is not None):`
Also, shouldn't we unset the value if it is not there? (I have no idea about this one, I'm not in the remembering state, just the asking questions state.)

Reply:
Me neither :D
State looks like this, so I'm counting on that the shape will be there, but the values are set to `None`. Maybe I could find more in OctoPrint documentation, about the state. Maybe we should at leat check if `data is not None` 
```json
{
  "state": {
    "text": "Cancelling",
    "flags": {
      "operational": True,
      "printing": True,
      "cancelling": True,
      "pausing": False,
      "resuming": False,
      "finishing": False,
      "closedOrError": False,
      "error": False,
      "paused": False,
      "ready": False,
      "sdReady": True
    }
  },
  "job": {
    "file": {
      "name": "M3D_Filament_Guide_triple.gco",
      "path": "M3D_Filament_Guide_triple.gco",
      "display": "M3D_Filament_Guide_triple.gco",
      "origin": "local",
      "size": 1105990,
      "date": 1450214458
    },
    "estimatedPrintTime": 998,
    "averagePrintTime": None,
    "lastPrintTime": None,
    "filament": {
      "tool0": {"length": 545, "volume": 1.3108786220955533},
      "tool1": {"length": 0, "volume": 0.0}
    },
    "user": "ra100"
  },
  "currentZ": None,
  "progress": {
    "completion": None,
    "filepos": None,
    "printTime": None,
    "printTimeLeft": None,
    "printTimeLeftOrigin": None
  },
  "offsets": {}
}
```

Reply:
I can let it go then, it's definitely better then before.